client/actions/userActions.js

```diff
@@ -0,0 +1,352 @@
+export function getAuthUser(id) {
+  return (dispatch) => {
+    return axios.get(`/users/${id}`).then((res) => {
+      dispatch(getAuthUserSuccess(res.data.user));
+    });
+  };
+}
+
+/**
+ * user update with admin authentication
+ * PUT /users/:id
+ * @param  {object} user [user data object to update]
+ * @param  {number} id   user id
+ * @return {object}      [api response]
+ */
+export function updateUserAdmin(user, id) {
+  return (dispatch) => {
+    return axios.put(`/users/${id}`, user).then(() => {
+      dispatch(loadUsers());
+    });
+  };
+}
+
+/**
+ * user update by account owner
+ * GET /users/:id
+ * @param  {object} user [user data object to update]
+ * @param  {number} id   user id
+ * @return {object}      [api response]
+ */
+export function updateUser(user, id) {
+  return (dispatch) => {
+    return axios.put(`/users/${id}`, user).then(() => {
+      dispatch(getAuthUser(id));
+    });
+  };
+}
+
+/**
+ *
+ *create a new user by admin
+ *POST /users/
+ * @export saveUser
+ * @param {any} user
+ * @returns {Object} json object
+ */
+export function saveUserAdmin(user) {
+  return (dispatch) => {
+    return axios.post('/users', user)
+    .then(() => {
+      dispatch(loadUsers());
+    });
+  };
+}
+
+/**
+ * get selected user details
+ * GET /users/:id
+ *
+ * @export getUser
+ * @param {any} id
+ * @returns {Object} json object
+ */
+export function getUser(id) {
+  return (dispatch) => {
+    return axios.get(`/users/${id}`)
+    .then((res) => {
+      dispatch(getUserSuccess(res.data.user.name));
+    });
+  };
+}
+
+/**
+ *
+ *
+ * @export getUser
+ * @param {any} id
+ * @returns {Object} json object
+ */
+export function deleteUser(id) {
+  return (dispatch) => {
+    return axios.delete(`/users/${id}`)
+    .then(() => {
+      dispatch(loadUsers());
+    });
+  };
+}
+
+/**
+ * isUserExists to check if passed parameter already exist in the db
+ * using the route GET /api/users/:identifier
+ *
+ * @export
+ * @param {any} identifier
+ * @returns {objeect} uuser
+ */
+export function isUserExists(identifier) {
+  return () => {
+    return axios.get(`/api/users/${identifier}`);
+  };
+}
+
+/**
+ * login user function
+ * set user token to local storage on successfull login
+ * and set headers for authorization
+ *
+ * @export
+ * @param {any} user
+ * @returns {any} data
+ */
+export function login(user) {
+  return (dispatch) => {
+    return axios.post('/users/login', user.user).then((res) => {
+      const token = res.data.token;
+      localStorage.setItem('jwtToken', token);
+      setAuthorizationToken(token);
+      axios.defaults.headers.common.Authorization = token;
+      dispatch(setCurrentUser(jwtDecode(token)));
+    });
+  };
+}
+
+/**
+ *
+ * logout user and clear token from localStorage
+ * @export
+ * @returns {any} data
+ */
+export function logout() {
+  return (dispatch) => {
+    localStorage.removeItem('jwtToken');
+    setAuthorizationToken(false);
+    dispatch(setCurrentUser({}));
+  };
+}
```

Comment:
Use the @export annotation for all your exported functions. You used in some and left some out. Also, make sure you are referring to the right function names